quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/TimefoldProcessorOverridePropertiesAtRuntimeTest.java

```diff
@@ -48,6 +48,9 @@ private static String getRequiredProperty(String name) {
             .overrideConfigKey("quarkus.timefold.solver.termination.best-score-limit", "0")
             .overrideConfigKey("quarkus.timefold.solver.move-thread-count", "4")
             .overrideConfigKey("quarkus.timefold.solver-manager.parallel-solver-count", "1")
+            .overrideConfigKey("quarkus.timefold.solver.termination.diminished-returns.enabled", "false")
```

Comment:
Arguably this is not necessary, if either of the other two properties is present?

Reply:
The other two properties are optional, which make it hard to use the defaults unless there a third property.
Additionally, if you want to use diminishing returns by default, but have the option to disable it, you need to somehow delete the defined properties if there is no enabled property.

Reply:
I see the reason for the property existing - I'm just arguing you don't need to specify it, if you specify any of the others.

Does it get confusing as you switch from one to the other? Probably. But how often does that actually happen?